### DISCUSSION TITLE
Could someone point me in the right direction on getting the HTTP status portion going?

### DISCUSSION THREAD

**Discussion by vPeteWalker** [lloesche/valheim-server-docker#256] [Q&A]
Hi folks. OK everything else is working, and setup via docker compose. I am hoping maybe someone can share some configs, or prereqs, or whatever else on getting this going I think I'd be good to go. I am using an .env file, and put the entries in, but I don't know if the examples list should be put in exactly the same, or modified for my config?

If I understand what it does, just a web page I can view status of the server. Even if it just allows me to see if anyone else is on before I do maintenance with the docker host itself would be a huge help. Thanks very much folks.

**Comment by lloesche** [comment]
@vPeteWalker you will have to expose the status http server port or use the generated document in some other way. By default the port inside the container is standard http `80`. But you will likely want to map it to something like `9080` on your host (`-p 9080:80`) or in your compose file:
```
    ports:
      - "2456-2457:2456-2457/udp"
      - "9001:9001/tcp"
      - "9080:80/tcp"
```

You can also run the command `valheim-is-idle` to figure out it there are players connected. It exits `0` if the server is idle or not running Ie. when no players are connected.

**Reply by vPeteWalker** [reply]
Thank you man. I really appreciate you taking the time here. What am I doing wrong here? I cannot hit the site via internal or external on port 9080. Is there something special in the URL I'm missing? I'm using http://<IP-ADDRESS>:9080 and it's not working.

Also, would I type that valheim-is-idle command in an interactive session for the container?

My config:

STATUS_HTTP=true
STATUS_HTTP_PORT=9080
STATUS_HTTP_CONF=/config/httpd.conf
STATUS_HTTP_HTDOCS=/opt/valheim/htdocs

Port 9080 is opened on my firewall, and forwarded to the Valheim host.

Port 9080 configured in my docker-compose for Valheim, for TCP. I don't know what 9001 is used for, but I am forwarding that as well for the time being:

ports:
      - 2455-2458:2455-2458/udp
      - 9001:9001/tcp
      - 9080:9080/tcp

**Reply by ste032** [reply]
You need to query this url instead: http://your-ip:9080/status.json

Also, port 9001 is used for the Supervisor process (refer to the documentation for an indepth explanation): DO NOT FORWARD THAT TO THE INTERNET, it is meant to be used locally.

edit: fixed missing // in url

**Reply by vPeteWalker** [reply]
Thanks @ste032. I removed the 9001 port forwarding. I cannot reach that site, internally or externally. Do you have this working? If so, can you share your config? Thank you once again.

**Reply by ste032** [reply]
Yeah, it's all working on my end:

valheim.env
```
...
SERVER_PUBLIC=true   (you also need this one)
STATUS_HTTP=true
STATUS_HTTP_PORT=8181
```

docker-compose.yaml
```
...
ports:
      - "2456-2457:2456-2457/udp"
      - "9001:9001/tcp"
      - "8181:8181/tcp"
```

I can access the HTTP status page both internally (local IP) and externally (public IP\dns).

`http://<IP>:8181/status.json`

**Reply by vPeteWalker** [reply]
Okay, excellent. All I did was update (since Valheim got a patch today), and now it's working, internally and externally. What the hell?

**Reply by ste032** [reply]
You might have had the correct configuration from the start, but one explanation could be that you were checking the wrong url initially (missing the /status.json), then you tried what I suggested in my first reply that was also wrong due to a typo (missing //), then tried with the url in my last post which ultimately was correct!

Anyway, glad it's resolved.

**Reply by vPeteWalker** [reply]
Appreciate everyone's help! This is a great little solution.

**Comment by troubleshootme** [comment]
For what it's worth, this is my ugly ```index.html``` file I placed in the ```htdocs``` folder to make status easier to read. 
Don't forget to adjust the fetch params to match your status.json url.
```
<!DOCTYPE html>
<html>
  <head>
    <title>Status Table</title>
    <style>
      table,
      th,
      td {
        border: 1px solid black;
        border-collapse: collapse;
        padding: 5px;
      }
    </style>
  </head>
  <body>
    <h1>Status Table</h1>
    <table>
      <thead>
        <tr>
          <th>Field Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        <script>

<!-- NOTE: Set the url below to your status.json file-->

          fetch('http://your.path.to:port/status.json')
            .then(response => response.json())
            .then(data => {
              Object.keys(data).forEach(key => {
                const formattedKey = key.split('_').join(' ');
                const value = data[key];
                const row = `<tr><td>${formattedKey}</td><td>${value}</td></tr>`;
                document.querySelector('tbody').insertAdjacentHTML('beforeend', row);
              });
            });
        </script>
      </tbody>
    </table>
  </body>
</html>
```
